app/controllers/admin/content/events.js

```diff
@@ -0,0 +1,73 @@
+import Controller from '@ember/controller';
+import { camelCase, startCase } from 'lodash';
+
+export default Controller.extend({
+  actions: {
+    async loadSubTopics(topic) {
+      this.set('isLoading', true);
+      try {
+        this.set('subTopics', await topic.get('subTopics'));
+      } catch (e) {
+        this.notify.error(this.get('l10n').t('An unexpected error has occurred. SubTopics not loaded.'));
+      } finally {
+        this.set('isLoading', false);
+      }
+    },
+    openModalFor(modelInstanceOrName) {
+      const modelName = typeof modelInstanceOrName === 'string' ? modelInstanceOrName : modelInstanceOrName.modelName;
+      const modelInstance = typeof modelInstanceOrName === 'string' ? this.store.createRecord(modelInstanceOrName) : modelInstanceOrName;
+      const camelCasedValue = camelCase(modelName);
+      this.set(camelCasedValue, modelInstance);
+      switch (camelCasedValue) {
+        case 'eventType':
+          this.set('isNewEventTypeModalOpen', true);
```

Comment:
Like I said before, these properties (`isNewEventTypeModalOpen` and such) are not needed. You can just directly pass in `eventType` to `isOpen`. That would be enough to control the model.